src/components/activities/MimicStats.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import type { LinearProgressProps } from '@mui/material';
+import { LinearProgress, Typography } from '@mui/material';
+import { Grid } from '@mui/material';
+import { Box } from '@mui/system';
+
+import type { GameResponse } from 'types/gameResponse.type';
+import { MimicResponseValue } from 'types/mimicResponse.type';
+import type { User } from 'types/user.type';
+
+import { Flag } from '../Flag';
+
+function LinearProgressWithLabel(props: LinearProgressProps & { value: number }) {
+  return (
+    <Box display="flex" alignItems="center">
+      <Box width="100%" mr={1}>
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box minWidth={35}>
+        <Typography variant="body2" color="textSecondary">{`${Math.round(props.value)}%`}</Typography>
+      </Box>
+    </Box>
+  );
+}
+
+interface mimicStatProps {
+  gameResponses: GameResponse[];
+  choices: number[];
+  country: string;
+  users: User[];
+  userMap: Record<number, number>;
+}
+
+export const MimicStats = ({ gameResponses, choices, country, userMap, users }: mimicStatProps) => {
+  const countryResponses = React.useMemo(() => {
+    return gameResponses.filter((responseGame) => users[userMap[responseGame.userId]]?.country.isoCode === country);
+  }, [country, gameResponses, userMap, users]);
+  const responseCount = countryResponses.length;
+  const stats = React.useMemo(() => {
+    return [
+      (countryResponses.filter((response) => response.value === MimicResponseValue.SIGNIFICATION).length * 100) / responseCount,
+      (countryResponses.filter((response) => response.value === MimicResponseValue.FAKE_SIGNIFICATION_1).length * 100) / responseCount,
+      (countryResponses.filter((response) => response.value === MimicResponseValue.FAKE_SIGNIFICATION_2).length * 100) / responseCount,
+    ];
+  }, [countryResponses, responseCount]);
+  const responseChoices = React.useMemo(() => {
+    return choices;
+  }, [choices]);
+
+  return (
+    <Grid item xs={5} md={5}>
+      <span style={{ paddingRight: '0.5rem' }}>{responseCount !== 0 ? 'Réponses de vos Pélicopains' : 'Pas de réponses de vos Pélicopains'}</span>
+      <Flag country={country} />
+      {responseCount > 0 && responseChoices && (
+        <>
+          {responseChoices.map((responseChoice) => {
+            if (responseChoice === 0) {
+              return <LinearProgressWithLabel value={stats[0]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            } else if (responseChoice === 1) {
+              return <LinearProgressWithLabel value={stats[1]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            } else {
+              return <LinearProgressWithLabel value={stats[2]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            }
+          })}
```

Comment:
**nit:** You can replace this if/else by using directly responseChoice as the array index!

**Suggestion:**
```tsx
{responseChoices.map((responseChoice) => (
    <LinearProgressWithLabel value={stats[responseChoice]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />
))}
```

Reply:
It is indeed cleaner. I had to add key prop though.

Reply:
It is indeed a cleaner way. I had to add  a key prop though.